src/python/dxpy/__init__.py

```diff
@@ -609,6 +609,13 @@ def ensure_ascii(i):
 
                 response = pool_manager.request(_method, encoded_url, headers=_headers, body=body,
                                                 timeout=timeout, retries=False, **kwargs)
+                # Handle redirection manually for symlink files
+                if response.status in range(300, 399):
+                    url = response.headers.get('Location')
+                    if url:
+                        # Make a new request to the URL specified in the Location header
+                        response = pool_manager.request(_method, url, headers=_headers, body=body,
```

Comment:
is there a chance for multiple redirects?

Reply:
I don't think so, we've tested with very large multipart and single part files and it should only need to redirect on each call to nginx.